event-based-simulators/extras/ExportProfileData.py

```diff
@@ -0,0 +1,192 @@
+import base64,json, logging, os, sys, requests
+
+import ArgumentsAndCredentialsHandler
+import Environment
+
+from datetime import datetime, timedelta, timezone
+
+####################################################
+if not os.path.exists('logs'):
+    os.makedirs('logs')
+logger = logging.getLogger('ExportImportProfileData')
+relativeFilePath = f"logs\{datetime.strftime(datetime.now(), '%Y%m%d%H%M%S_%f')}.log"
+filePath = os.path.join(os.path.dirname(__file__), relativeFilePath)
+hdlr = logging.FileHandler(filePath)
+formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+hdlr.setFormatter(formatter)
+logger.addHandler(hdlr)
+logger.setLevel(logging.DEBUG)
+####################################################
+# Setup for additional API request message
+user_and_pass_bytes = base64.b64encode(
+    (Environment.C8Y_TENANT + "/" + Environment.C8Y_USER + ':' + Environment.C8Y_PASSWORD).encode('ascii'))  # bytes
+user_and_pass = user_and_pass_bytes.decode('ascii')  # decode to str
+
+C8Y_HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Authorization': 'Basic ' + user_and_pass
+}
+
+
+#####################################################
+
+
+def exportAllProfileDataFromChildDevices(c8y, DATA_TYPE, createFrom, createTo):
+    # Loop through the list of device in Device management
+    try:
+        deviceManagedObject = c8y.device_inventory.select(type="c8y_EventBasedSimulator")
+    except:
+        logger.error(
+            "Connection to Cumulocity platform failed. Check your required parameters in environment file again")
+        sys.exit()
+
+    for device in deviceManagedObject:
+        logger.info(f"Found device '{device.name}', id: #{device.id}, "
+                    f"owned by {device.owner}, number of children: {len(device.child_devices)}, type: {device.type}")
+        logger.info(f"List of {device.name}'s child devices: ")
+        for childDevice in device.child_devices:
```

Comment:
here we need to filter for child devices which have an external ID of type "c8y_EventBasedSimulatorProfile" (you might want to add this external ID manually in your tenant for testing; in the end, the oee-simulator MS will do that automatically in the future)